django/contrib/auth/management/__init__.py

```diff
@@ -108,6 +109,90 @@ def create_permissions(
             print("Adding permission '%s'" % perm)
 
 
+class RenamePermission(migrations.RunPython):
+    def __init__(self, app_label, old_model, new_model):
+        self.app_label = app_label
+        self.old_model = old_model
+        self.new_model = new_model
+        super(RenamePermission, self).__init__(
+            self.rename_forward, self.rename_backward
+        )
+
+    def _rename(self, apps, schema_editor, old_model, new_model):
+        ContentType = apps.get_model("contenttypes", "ContentType")
+        from django.contrib.auth.models import Permission  # real live model import
+
+        db = schema_editor.connection.alias
+        if not router.allow_migrate_model(db, Permission):
+            return
+
+        ctypes = ContentType.objects.filter(
+            app_label=self.app_label, model__icontains=old_model
+        )
+        permissions = Permission.objects.filter(
+            content_type_id__in=[ct.id for ct in ctypes]
+        )
+        for permission in permissions:
+            prefix = permission.codename.split("_")[0]
+            new_codename = f"{prefix}_{new_model}"
+            new_name = f"Can {prefix} {new_model}"
```

Comment:
```suggestion
            default_verbose_name = camel_case_to_spaces(new_model)
            new_name = f"Can {prefix} {default_verbose_name}"
```
I think the Permission name uses the model verbose_name which I'm not sure we can access but we can probably do a better job of building. This will need some test updates :+1: